### PR TITLE
For corefx testing, use a matching corefx repo version git hash

### PR DESCRIPTION
**Pull request by BruceForstall**
If the user doesn't specify a specific corefx commit hash to use in run-corefx-tests.py,
try to find the matching commit hash in the coreclr repro. If the
matching hash can't be found, use 'HEAD'.

We find the matching corefx commit hash by first parsing file
'dependencies.props' at the root of the coreclr repro, looking for this:

```
   <MicrosoftPrivateCoreFxNETCoreAppPackageVersion>4.5.0-preview1-26112-01</MicrosoftPrivateCoreFxNETCoreAppPackageVersion>
```

This determines the corefx package version that matches. Next, we look for the version.txt
file in the package cache, e.g.,
```
   <coreclr_root>\packages\microsoft.private.corefx.netcoreapp\4.5.0-preview1-26112-01\version.txt
```
The contents of this file is exactly the git commit hash
we need to use, e.g.:
```
   197a0699b08087ea85581679afdd9fd7b5c465c3
```
The version.txt file is created when the corefx package is restored,
which happens when doing one of:
   Windows: tests\runtests.cmd GenerateLayoutOnly
   non-Windows: build-test.sh generatelayoutonly

It would also be possible to not depend on the package already
being downloaded, but instead download the correct package here,
using the determined "MicrosoftPrivateCoreFxNETCoreAppPackageVersion"
package version, e.g.:
```
   https://dotnet.myget.org/F/dotnet-core/api/v2/package/Microsoft.Private.CoreFx.NETCoreApp/4.5.0-preview1-26112-01
```
and then extracting the ZIP archive to find the version.txt file.

This might get easier if the corefx commit hash is added directly to dependencies.props, as
discussed in https://github.com/dotnet/buildtools/issues/1141.

Change CI to build a "GenerateLayoutOnly" drop for corefx testing,
specifically to cause the corefx package to be restored, so we
can determine its version.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
cc @jkotas

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu x64 Checked corefx_baseline
@dotnet-bot test Windows_NT x64 Checked corefx_baseline

**Comment by jkotas** [comment]
There is CoreCLR/CoreFX breaking change in flight right now, so corefx_baseline jobs will fail. To make these CoreFX jobs always green, we would need to:

1. Let CoreFX build against older CoreCLR that it is known to build against successfully. My preference would be to download it as a binary drop instead of building it; but building it fresh locally will work too.
2. Combine the CoreFX build with the live CoreCLR.
3. Have exclusion list for temporarily failing CoreFX tests to compensate for breaking changes being pushed through the system.

More discussion about this is in https://github.com/dotnet/coreclr/issues/12782

File: netci.groovy [hunk 1]
```diff
@@ -1650,7 +1650,12 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
                     }
 
                     if (enableCorefxTesting) {
-                        buildOpts += ' skiptests';
+                        // We shouldn't need to build the tests. However, run-corefx-tests.py currently depends on having the restored corefx
+                        // package available, to determine the correct corefx version git commit hash, and we need to build the tests before
+                        // running "tests\\runtest.cmd GenerateLayoutOnly". So build the pri-0 tests to make this happen.
+                        //
+                        // buildOpts += ' skiptests';
```
Comment (jashook) [inline netci.groovy]: This seems like an expensive hack. To my knowledge runtest generateLayoutOnly does not require built tests to have run.

Comment (BruceForstall) [reply netci.groovy]: Yes, it is (relatively) expensive. But you do actually need to have built tests (on Windows) before GenerateLayoutOnly will work. I don't think it used to be that way.

**Pull request merged**